Big Data Assignment 1
Camila Licona Toledo - A00838736
August 25, 2026

In [18]:
import pandas as pd
import numpy as np

question 1

In [19]:
mis_datos = pd.read_csv("cars.csv")

In [20]:
mis_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
0,Acura,MDX,SUV,Asia,All,36945,33337,3.5,6.0,265.0,17.0,23.0,4451.0,106.0,189.0
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,21761,2.0,4.0,200.0,24.0,31.0,2778.0,101.0,172.0
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,24647,2.4,4.0,200.0,22.0,29.0,3230.0,105.0,183.0
3,Acura,TL 4dr,Sedan,Asia,Front,33195,30299,3.2,6.0,270.0,20.0,28.0,NaN,108.0,186.0
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,39014,3.5,6.0,225.0,18.0,24.0,3880.0,115.0,197.0


In [21]:
print("Rows and columns:", mis_datos.shape)

Rows and columns: (428, 15)


In [22]:
mis_datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 428 entries, 0 to 427
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Make         428 non-null    object 
 1   Model        428 non-null    object 
 2   Type         428 non-null    object 
 3   Origin       428 non-null    object 
 4   DriveTrain   428 non-null    object 
 5   MSRP         428 non-null    int64  
 6   Invoice      428 non-null    int64  
 7   EngineSize   425 non-null    float64
 8   Cylinders    425 non-null    float64
 9   Horsepower   423 non-null    float64
 10  MPG_City     426 non-null    float64
 11  MPG_Highway  426 non-null    float64
 12  Weight       423 non-null    float64
 13  Wheelbase    426 non-null    float64
 14  Length       427 non-null    float64
dtypes: float64(8), int64(2), object(5)
memory usage: 50.3+ KB


In [23]:
mis_datos.describe()

,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
count,428.000000,428.000000,425.000000,425.000000,423.000000,426.000000,426.000000,423.000000,426.000000,427.000000
mean,32774.855140,30014.700935,3.195294,5.811765,215.576832,20.065728,26.840376,3578.229314,108.166667,186.381733
std,19431.716674,17642.117750,1.111932,1.557802,71.978098,5.249574,5.749505,759.034549,8.327582,14.369110
min,10280.000000,9875.000000,1.300000,3.000000,73.000000,10.000000,12.000000,1850.000000,89.000000,143.000000
25%,20334.250000,18866.000000,2.300000,4.000000,165.000000,17.000000,24.000000,3103.000000,103.000000,178.000000
50%,27635.000000,25294.500000,3.000000,6.000000,210.000000,19.000000,26.000000,3476.000000,107.000000,187.000000
75%,39205.000000,35710.250000,3.900000,6.000000,252.500000,21.750000,29.000000,3978.500000,112.000000,194.000000
max,192465.000000,173560.000000,8.300000,12.000000,500.000000,60.000000,66.000000,7190.000000,144.000000,238.000000


The table has 428 cars and 15 columns. Five columns are text and the other ten are numbers. 

some columns show less than 428 non-null values, so those are the ones that have missing data.

question 2

In [24]:
mis_datos.isna().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     3
Cylinders      3
Horsepower     5
MPG_City       2
MPG_Highway    2
Weight         5
Wheelbase      2
Length         1
dtype: int64

In [25]:
mis_datos = mis_datos.fillna(mis_datos.mean(numeric_only=True))

print("Missing values after the fillna:", mis_datos.isna().sum().sum())

Missing values after the fillna: 0


There were 23 missing values in eight numeric columns. I replaced them with the mean of each column. 

question 3

In [26]:
cars_by_make = mis_datos["Make"].value_counts()
cars_by_make

Make
Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              20
Audi             19
Nissan           17
Honda            17
Chrysler         15
Volkswagen       15
Mitsubishi       13
Dodge            13
Volvo            12
Hyundai          12
Jaguar           12
Subaru           11
Kia              11
Pontiac          11
Mazda            11
Lexus            11
Buick             9
Mercury           9
Lincoln           9
Cadillac          8
GMC               8
Saturn            8
Suzuki            8
Infiniti          8
Acura             7
Saab              7
Porsche           7
Land Rover        3
Oldsmobile        3
Jeep              3
Isuzu             2
MINI              2
Scion             2
Hummer            1
Name: count, dtype: int64

In [27]:
print("More cars:", cars_by_make.idxmax(), "with", cars_by_make.max())
print("Less cars:", cars_by_make.idxmin(), "with", cars_by_make.min())

More cars: Toyota with 28
Less cars: Hummer with 1


Toyota is the manufacturer with more cars in the table, with 28, and Hummer is the one with less, with only 1. 

question 4

In [28]:
pd.crosstab(mis_datos["Origin"], mis_datos["DriveTrain"])

DriveTrain,All,Front,Rear
Origin,,,
Asia,34,99,25
Europe,36,37,50
USA,22,90,35


Asia has 99 front, 25 rear and 34 all. USA has 90 front, 35 rear and 22 all. Europe is the different one because it has 50 rear, more than the other two regions.

question 5 

In [29]:
print("Highest MSRP:", mis_datos["MSRP"].max())
print("Lowest MSRP:", mis_datos["MSRP"].min())
print("Average MPG_City:", round(mis_datos["MPG_City"].mean(), 2))
print("Average MPG_Highway:", round(mis_datos["MPG_Highway"].mean(), 2))

Highest MSRP: 192465
Lowest MSRP: 10280
Average MPG_City: 20.07
Average MPG_Highway: 26.84


The most expensive car costs 192,465 dollars and the cheapest one 10,280. The average is 20.07 miles per gallon in the city and 26.84 on the highway.

question 6 

In [30]:
columnas = ["Make", "Model", "Type", "Origin", "DriveTrain",
            "MSRP", "Weight", "Wheelbase", "Length"]

mis_nuevos_datos = mis_datos[columnas][mis_datos["DriveTrain"] == "Front"].copy()

print("Size of mis_nuevos_datos:", mis_nuevos_datos.shape)

Size of mis_nuevos_datos: (226, 9)


From the 428 cars, 226 have front wheel drive.

question 7

In [31]:
mis_nuevos_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Weight,Wheelbase,Length
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,2778.000000,101.0,172.0
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,3230.000000,105.0,183.0
3,Acura,TL 4dr,Sedan,Asia,Front,33195,3578.229314,108.0,186.0
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,3880.000000,115.0,197.0
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,3893.000000,115.0,197.0


In [32]:
mis_nuevos_datos["Weight_kilo"] = mis_nuevos_datos["Weight"] * 0.453592
mis_nuevos_datos["Wheelbase_cm"] = mis_nuevos_datos["Wheelbase"] * 2.54
mis_nuevos_datos["Length_cm"] = mis_nuevos_datos["Length"] * 2.54

mis_nuevos_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Weight,Wheelbase,Length,Weight_kilo,Wheelbase_cm,Length_cm
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,2778.000000,101.0,172.0,1260.078576,256.54,436.88
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,3230.000000,105.0,183.0,1465.102160,266.70,464.82
3,Acura,TL 4dr,Sedan,Asia,Front,33195,3578.229314,108.0,186.0,1623.056191,274.32,472.44
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,3880.000000,115.0,197.0,1759.936960,292.10,500.38
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,3893.000000,115.0,197.0,1765.833656,292.10,500.38


One pound is 0.453592 kilos and one inch is 2.54 centimetres. The average weight is around 1,500 kilos and the average length around 470 centimetres, which is 4.7 metres.

question 8

In [33]:
def opinion(precio):
    if precio > 60000:
        return "Expensive"
    elif precio >= 30000:
        return "Normal"
    else:
        return "Inexpensive"

mis_nuevos_datos["opinion"] = mis_nuevos_datos["MSRP"].apply(opinion)

mis_nuevos_datos["opinion"].value_counts()

opinion
Inexpensive    175
Normal          48
Expensive        3
Name: count, dtype: int64

In [34]:
mis_nuevos_datos.head(10)

,Make,Model,Type,Origin,DriveTrain,MSRP,Weight,Wheelbase,Length,Weight_kilo,Wheelbase_cm,Length_cm,opinion
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,2778.000000,101.0,172.0,1260.078576,256.54,436.88,Inexpensive
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,3230.000000,105.0,183.0,1465.102160,266.70,464.82,Inexpensive
3,Acura,TL 4dr,Sedan,Asia,Front,33195,3578.229314,108.0,186.0,1623.056191,274.32,472.44,Normal
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,3880.000000,115.0,197.0,1759.936960,292.10,500.38,Normal
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,3893.000000,115.0,197.0,1765.833656,292.10,500.38,Normal
7,Audi,A4 1.8T 4dr,Sedan,Europe,Front,25940,3252.000000,104.0,179.0,1475.081184,264.16,454.66,Inexpensive
8,Audi,A41.8T convertible 2dr,Sedan,Europe,Front,35940,3638.000000,105.0,180.0,1650.167696,266.70,457.20,Normal
9,Audi,A4 3.0 4dr,Sedan,Europe,Front,31840,3462.000000,104.0,179.0,1570.335504,264.16,454.66,Normal
12,Audi,A6 3.0 4dr,Sedan,Europe,Front,36640,3561.000000,109.0,192.0,1615.241112,276.86,487.68,Normal
14,Audi,A4 3.0 convertible 2dr,Sedan,Europe,Front,42490,3814.000000,105.0,180.0,1729.999888,266.70,457.20,Normal


From the 226 cars with front wheel drive, 175 are Inexpensive, 48 are Normal and only 3 are Expensive.